src/components/groups/groupsService.js

```diff
@@ -140,47 +132,41 @@
                     });
                 };
 
-                function _getAuthSrv(){
-                    return $injector.get(AuthSrvName);
-                }
-
-                function _getStorage(){
-                    return $injector.get(StorageSrvName);
-                }
-
-                function _getGroupPath(){
-                    return _getStorage().variables.appUserSpacePath + '/groups';
-                }
-
-                function addGroupListener() {
-                    var authData = _getAuthSrv().getAuth();
+                GroupsService.addGroupsListener = function (eventName, callback) {
+                    var authData = authSrv.getAuth();
                     if (authData && authData.uid) {
-                        var ref = new GroupsFirebaseListener(authData.uid);
-                        ref.on('child_added', groupsChildAdded);
-                        ref.on('child_removed', groupsChildRemoved);
+                        var fullPath = ENV.fbDataEndPoint + ENV.firebaseAppScopeName + '/' + GROUPS_PATH;
+                        var groupsFullPath = fullPath.replace('$$uid', authData.uid);
+                        var ref = new Firebase(groupsFullPath);
+
+                        if (angular.isFunction(callback)) {
+                            ref.on(eventName, callback);
+                        }
                     }
-                }
+                };
 
-                function groupsChildAdded(dataSnapshot) {
-                    $timeout(function () {
-                        GroupsService.groups[dataSnapshot.key()] = dataSnapshot.val();
-                    });
-                }
+                GroupsService.removeGroupsListener = function (eventName, callback) {
```

Comment:
addGroupsListener and remove function are the same.You can use one func with param